experimental/dds/tree2/src/simple-tree/toMapTree.ts

```diff
@@ -160,11 +184,14 @@ function valueToMapTree(
 	context: TreeDataContext,
 	typeSet: TreeTypeSet,
 ): MapTree {
-	const type = getType(value, context, typeSet);
-	const schema = getSchema(context, type);
-	assert(allowsValue(schema.leafValue, value), "Unsupported schema for provided primitive.");
+	const mappedValue = mapValueWithFallbacks(value, typeSet);
```

Comment:
This change isn't strictly required, but validating the value post-fallback-application is a nice sanity check